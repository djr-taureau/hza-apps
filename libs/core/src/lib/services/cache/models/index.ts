// start:ng42.barrel
export * from './cache-entry';
export * from './cache';
// end:ng42.barrel

