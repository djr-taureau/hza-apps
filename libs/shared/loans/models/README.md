# shared-loans-models

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-loans-models` to execute the unit tests.
