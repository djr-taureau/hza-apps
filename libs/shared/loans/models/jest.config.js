module.exports = {
  name: 'shared-loans-models',
  preset: '../../../../jest.config.js',
  coverageDirectory: '../../../../coverage/libs/shared/loans/models',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
