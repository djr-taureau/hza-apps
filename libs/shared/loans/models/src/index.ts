export * from './lib/models/loan.model';
