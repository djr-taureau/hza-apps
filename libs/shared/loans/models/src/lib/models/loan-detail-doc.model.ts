export interface LoanDetailDoc {
	SourceId: number;
	LoanNumber: string;
	InvestorName: string;
	BorrowerPrimaryFullName: string;
	BorrowerPrimaryFirstName: string;
	BorrowerPrimaryLastName: string;
	BorrowerSSN: string;
	BorrowerPrimaryEmailAddress: string;
	BorrowerPrimaryPhoneNumber: string;
	BorrowerFICOScore: number;
	FICOScoreDate: Date;
	BorrowerSecondaryFullName: string;
	BorrowerCoSSN: string;
	BorrowerSecondaryPhoneNumber: string;
	BorrowerSecondaryPhoneNumberMSP: string;
	PropertyStreet: string;
	PropertyCity: string;
	PropertyStateCode: string;
	PropertyZipCode: string;
	MailingStreet: string;
	MailingCity: string;
	MailingState: string;
	MailingZipCode: string;
	BrokerPriceOpinionValue: number;
	BrokerPriceOpinionDate: Date;
	BasisAmount: number;
	CurrentLegalBalanceAmt: number;
	UnpaidBalance: number;
	SuspenseBalanceAmt: number;
	NextPaymentDueDate: Date;
	LastPaymentDate: Date;
	ProjectedResolutionType: number;
	ProjectedResolutionDate: Date;
	EscrowBalance: number;
	LoanType: string;
	FCAttorneyPhoneNumber: string;
	PaymentHistory: string;
	MonthlyInsurance: number;
	MonthlyPIAmount: number;
	PITIAmt: number;
	AutoACHInd: string;
	AutoACHDate: Date;
	CurrentMonthlyTaxAmt: number;
	CurrentMonthlyMortgageInsAmt: number;
	EscrowProjectedShortageAmt: number;
	AttorneyOnFileInd: boolean;
	LastSkipTraceDate: Date;
	LastDoorKnockDate: string;
	NoContactLetterSentDate: Date;
	DefaultReasonDescr: string;
	PriorDefaultReasonDescr: string;
	PaymentPlanNextDueDate: Date;
	OTDDate: Date;
	OTDAmount: number;
	PaymentChangeDate: Date;
	PaymentChangeAmt: number;
	PaymentPlanType: string;
	PaymentPlanPaymentAmt: number;
	PaymentPlanNumberOfPayments: number;
	PaymentPlanFinalPaymentDate: Date;
	LMLastBAFSentDate: Date;
	LossMitRemovalDate: Date;
	BankruptcyChapterDescr: string;
	BankruptcySurrender: string;
	BankruptcyDischargedInd: string;
	Terminated: string;
	BankruptcyRemovalDate: Date;
	FCProcessor: string;
	FCAttorney: string;
	DemandLetterSentDate: Date;
	DemandLetterExpirationDate: Date;
	MostRecentEvalSentDate: Date;
	InvestorLoanNumber: string;
	LatestValuationValue: number;
	LatestValuationDate: Date;
	ContactBoardDate: Date;
	DeferredBalance: number;
	LegalBalance: number;
	TaxLastPaidPayeeName: string;
	TaxLastPaidAmt: number;
	TaxLastPaidDate: Date;
	TaxNextDueAmt: number;
	TaxNextDueDate: Date;
	HazardLastPaidPayeeName: string;
	HazardLastPaidAmt: number;
	HazardLastPaidDate: Date;
	HazardNextDueAmt: number;
	HazardNextDueDate: Date;
	HazardPolicyNumber: string;
	MILastPaidPayeeName: string;
	MILastPaidDate: Date;
	MILastPaidAmt: number;
	MINextDueAmt: number;
	MINextDueDate: Date;
	NextPmtDueDate: Date;
	PropertyInspectionDate: Date;
	PropertyConditionDescr: string;
	AnnualTaxes: number;
	LMSetupDate: Date;
	BKIsActiveInd: string;
	LMStatusCode: string;
	FCStatusCode: string;
	FCIsActiveInd: string;
	FCSaleScheduledDate: Date;
	CurrentMonthlyHazardInsAmt: number;
	LocationOccupancyStatusDescr: string;
	TaxNextDue: Date;
	ActiveLM: boolean;
	WaterfallStatus: string;
	LastApplicationReceived: Date;
	CompleteApplicationReceived: Date;
	IsUSSoldierInd: string;
	FCHoldFlag: string;
	FCHoldReason: string;
	FCHoldStartDate: Date;
	FCHoldProjectedEndDate: Date;
	FCTotalHoldDays: number;
	DaysInFC: number;
	DaysInFCAtFay: number;
	DaysInFCAtFayNotOnHold: number;
	ContestedFlag: boolean;
	FCReferralDate: Date;
	FirstLegalDate: Date;
	FCServiceCompletedDate: Date;
	FCAnswerFiledDate: Date;
	FCJudgementDate: Date;
	FCSaleActualDate: Date;
	FCBidAmount: number;
	FCSuccessfulBidAmount: number;
	FCThirdPartyFlag: string;
	FCAttorneyEmail: string;
	FCRemovalReason: string;
	BKFilingDate: Date;
	BKCaseNumber: number;
	DebtorAttorney: string;
	BKCourt: string;
	BKSetupDate: Date;
	BKAttorney: string;
	BKAttorneyPhoneNumber: string;
	BKAttorneyEmail: string;
	MFRReferredDate: Date;
	MFRFilingDeadline: Date;
	MFRFiledDate: Date;
	InRemMFRReferredDate: Date;
	MFRHearingDate: Date;
	MFRGrantedFlag: boolean;
	MFRGrantedDate: Date;
	MFRDeniedDate: Date;
	POCBarDate: Date;
	POCFiledDate: Date;
	TOCFiledDate: Date;
	TrusteePayAllFlag: string;
	PostPetitionDueDate: Date;
	CramdownFlag: boolean;
	LitigationFlag: boolean;
	AgreedOrderDefault: string;
	AgreedOrderDefaultReferred: string;
	PlanReviewReferred: string;
	StatementReconciliationCompleteDate: Date;
	ConfirmationHearingDate: Date;
	PlanConfirmedDate: Date;
	BKDischargeDate: Date;
	BKDismissalDate: Date;
	InvestorHeader: string;
	InvestorPortfolioName: string;
	PriorServicerLoanID: string;
	AmqLoanType: string;
	CurrentInterestRatePct: number;
	RecoverCorpAdvanceBalance: number;
	NonRecCorpAdvanceBalance: number;
	TotalCorpAdvanceBalance: number;
	FirstDueDate: '2007-06-25T00:00:00';
	EscAdvanceBalance: number;
	LienPosition: string;
	NoteDate: '2007-05-25T00:00:00';
	MaturityDate: '2058-08-01T00:00:00';
	OriginalBalance: number;
	ModificationFlag: boolean;
	ModificationDate: Date;
	LiquidationReason: string;
	AmqLiquidationDate: Date;
	ServiceTransferFlag: boolean;
	PaymentHistory36Months: string;
	PriorModFlag: boolean;
	PriorModDate: Date;
	DaysinStage: string;
	LastDocumentReceived: Date;
	Template: string;
	LossMitRemovalReason: string;
	AppealTask: string;
	AppealTaskStatus: string;
	AppealTaskStartDate: Date;
	AppealTaskCloseDate: Date;
	InteriorBPORequested: boolean;
	InteriorBPOReceived: boolean;
	DILTitleCurativeInitiated: boolean;
	DILTitleClear: boolean;
	DILInvestorApproval: boolean;
	DILDocsToBorrower: boolean;
	DILDocsExecuted: boolean;
	ReceivedTPA: boolean;
	ConfirmPropertyListed: boolean;
	ApprovedListPrice: number;
	ApprovedListPriceDate: null;
	ReceivedSaleContract: boolean;
	ShortSaleInvestorApprovalReceived: boolean;
	ShortSaleApprovalLetterSent: boolean;
	ShortSaleScheduledClosingDate: Date;
	ShortSaleActualClosingDate: Date;
	ShortSaleContractPrice: number;
	ShortSaleNetProceeds: number;
	LoType: string;
	AM: string;
	AMPhone: string;
	AMEmail: string;
	AMSupervisor: string;
	AMMessage: string;
	TitleEmployee: string;
	Location: string;
	TopSegmentName: string;
	TopSubsegmentName: string;
	CallWindowStart: string;
	CallWindowEnd: string;
	IsInCallWindow: boolean;
	IsProcessed: boolean;
	CampaignDesc: string;
	PaymentLastAmt: number;
	LossMitigationType: string;
	DelinquencyDays: string;
	EscrowRestrictedBalance: number;
	ND7Flag: number;
	BKFlag: string;
	ConditionalMessage: string;
	CallUrgency: string;
	ForeignAddress: boolean;
	AssignedUserId: string;
	DelinquencyClassCode: string;
	BKStatusCode: string;
	HiType: string;
	ReferralOriginations: string;
	ReferralInsurance: string;
	FollowUpDate: Date;
	LastContactDate: Date;
	LastAttemptDate: Date;
	Segmentation: string;
	VMAppPrimary: boolean;
	VMAppSecondary: boolean;
	AltLanguage: string;
	AttorneyOnFile: boolean;
	EmailAuth: boolean;
	EmailId: string;
	ReasonForDefault: string;
	ResolutionType: string;
	VlmOccupancy: string;
	VlmOccupancyDate: Date;
	PTPAmount: number;
	PTPDate: Date;
	PTPAffirm: string;
	ActiveREOCode: string;
	LateFeeBalance: number;
	NSFFeeBalance: number;
	TotalAmountDue: number;
	BankruptcyRemovalCode: string;
	LitigationStatusCode: string;
	FEMAIndicator: string;
	LastOrigination: string;
	LastInsurance: string;
	ModExplanationLoanOfficer: string;
	ModExplanationPerformed: boolean;
	OptionsOffered: boolean;
}