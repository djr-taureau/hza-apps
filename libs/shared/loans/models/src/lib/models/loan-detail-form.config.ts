export interface FormBuilderModel {
	displayName: string;
	fieldName: string;
	fieldType: string;
	cssSelector: string;
	templateCssSelector: string;
	fxFlexAlignField?: string;
}

export const docLoanDetailBorrower: FormBuilderModel[] = [
	{
		displayName: 'Borrower',
		fieldName: 'BorrowerPrimaryLastName',
		fieldType: 'input',
		cssSelector: 'value',
		templateCssSelector: 'x-wide-disabled right'
	},
    	{
		displayName: 'Borrower SSN',
		fieldName: 'BorrowerSSN',
		fieldType: 'input',
		cssSelector: 'value',
		templateCssSelector: 'medium-disabled long-date right',
		fxFlexAlignField: 'end center'
	}
];
export const docLoanDetailCoBorrower: FormBuilderModel[] = [
    {
		displayName: 'Co Borrower',
		fieldName: 'BorrowerSecondaryFullName',
		fieldType: 'input',
		cssSelector: 'value',
		templateCssSelector: 'x-wide-disabled right',
		fxFlexAlignField: 'end center'
	},
        {
		displayName: 'Co Borrower SSN',
		fieldName: 'BorrowerCoSSN',
		fieldType: 'input',
		cssSelector: 'value',
		templateCssSelector: 'medium-disabled long-date right'
	},
];

export const docLoanDetailInvestor: FormBuilderModel[] = [
    {
		displayName: 'Investor',
		fieldName: 'InvestorName',
		fieldType: 'input',
		cssSelector: 'value',
		templateCssSelector: 'x-wide-disabled right',
		fxFlexAlignField: 'end center'
	}
];

export const docLoanDetailProperty: FormBuilderModel[] = [
    {
		displayName: 'Property',
		fieldName: 'PropertyStreet',
		fieldType: 'textarea',
		cssSelector: 'value',
		templateCssSelector: 'x-wide-disabled',
		fxFlexAlignField: 'end center'
	}
];
