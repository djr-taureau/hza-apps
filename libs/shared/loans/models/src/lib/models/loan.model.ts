export interface Loan {
	loanNumber: string;
	sourceID: number;
	source: string;
	borrower: string;
	coBorrower: string;
	borrowerPrimarySSN: string;
	borrowerSecondarySSN: string;
	borrowerPrimaryEmailAddress: string;
	borrowerPrimaryPhoneNumber: string;
	borrowerSecondaryPhoneNumber: string;
	propertyAddress1: string;
	propertyCity: string;
}

export interface LoanDetail {
	borrower: string;
	borrowerPrimarySSN: string;
	coBorrower: string;
	borrowerSecondarySSN: string;
	propertyAddress1: string;
	propertyCity: string;
}

export function toLoanFormValue<T extends LoanDetail>(data: T): LoanDetail {
	return {
		borrower: data.borrower,
		borrowerPrimarySSN: data.borrowerPrimarySSN,
		coBorrower: data.coBorrower,
		borrowerSecondarySSN: data.borrowerSecondarySSN,
		propertyAddress1: data.propertyAddress1,
		propertyCity: data.propertyCity
	};
}

export const mockLoan1: Loan = {
	sourceID: 1,
	source: 'Fay',
	loanNumber: '0000015864',
	borrower: 'JEFFERY BARRON',
	coBorrower: 'MICHELLE BARRON',
	borrowerPrimarySSN: 'xxx-xx-6553',
	borrowerSecondarySSN: 'xxx-xx-7172',
	borrowerPrimaryEmailAddress: 'JMPROTEK@MTAONLINE.NET',
	borrowerPrimaryPhoneNumber: '9078926021',
	borrowerSecondaryPhoneNumber: null,
	propertyAddress1: '11765W BALLYSHANNON',
	propertyCity: 'HOUSTON'
};

export const mockLoan2: Loan = {
	sourceID: 1,
	source: 'Fay',
	loanNumber: '0000016534',
	borrower: 'MIGUEL RIOS',
	coBorrower: null,
	borrowerPrimarySSN: 'xxx-xx-2458',
	borrowerSecondarySSN: '',
	borrowerPrimaryEmailAddress: null,
	borrowerPrimaryPhoneNumber: '7132050366',
	borrowerSecondaryPhoneNumber: '7136579592',
	propertyAddress1: '11111  CLARK RD',
	propertyCity: 'HOUSTON'
};
