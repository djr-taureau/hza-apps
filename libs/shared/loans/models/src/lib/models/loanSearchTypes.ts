export interface LoanQuery {
	company: string;
	loan: string;
	loanSearch: string;
}

export const defaultQuery: LoanQuery = {
	company: '0',
	loan: 'loan',
	loanSearch: ''
};

export const mockLoanQuery: LoanQuery = {
	company: '0',
	loan: 'city',
	loanSearch: 'Houston'
};



export function toSearchFormValue<T extends LoanQuery>(data: T): LoanQuery {
  return {
    company: data.company,
    loan: data.loan,
    loanSearch: data.loanSearch,
  }
}

export function searchMessages(value: string): string {
	let message: string = '';
	switch (value) {
		case 'loan':
			message = 'Full loan number must be provided (leading zeros are optional)';
			break;
		case 'borrower':
			message = 'Search by the first few letters of first and/or last name';
			break;
		case 'coborrower':
			message = 'Search by the first few letters of first and/or last name';
			break;
		case 'ssn':
			message = 'Search by the last four  (or more) digits of the SSN';
			break;
		case 'email':
			message = 'Search by the first few characters of email address';
			break;
		case 'phone':
			message = 'Search by the last four or more digits of phone number';
			break;
		case 'address':
			message = 'Search by the first few numbers or letters of addesss';
			break;
		case 'city':
			message = 'Search by the first few letters of city';
			break;
		case 'zip':
			message = 'Full five digits of zip code must be provided';
			break;
		default:
			message = '';
			break;
	}

	return message;
}
