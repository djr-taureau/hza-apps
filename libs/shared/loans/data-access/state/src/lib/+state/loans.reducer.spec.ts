import { Loan, mockLoan1, mockLoan2, LoanQuery, mockLoanQuery, LoanDetailDoc, mockLoanDetailDoc } from '@hza/shared/loans/models';
import * as fromLoans from './loans.reducer';
import * as LoansActions from './loans.actions';
import * as loansQuery from '.';

describe('LoansReducer', () => {
	const loan1 = mockLoan1;
	const loan2 = mockLoan2;

	const loansInitialState: fromLoans.LoansState = {
		ids: [ loan1.loanNumber, loan2.loanNumber ],
		entities: {
			[loan1.loanNumber]: loan1,
			[loan2.loanNumber]: loan2
		},
		selectedLoanId: null,
		isLoading: false,
		loaded: true,
		loanQuery: null,
		loanDetail: null
	};

	describe('LOANS_LOAD_SUCCESS', () => {
		type LoansActions = typeof LoansActions.loadLoansSuccess;
		function noExistingLoans(action: LoansActions, loansInitialState: any, loans: Loan[]) {
			const createAction = action({ loans });

			const result = fromLoans.loansReducer(loansInitialState, createAction);

			expect(result).toMatchSnapshot();
		}

		it('should add all loans in the payload when none exist', () => {
			noExistingLoans(LoansActions.loadLoansSuccess, loansInitialState, [ loan1, loan2 ]);
		});
	});
	
	
		describe('LOAD LOAN DETAIL Success', () => {
		type LoansActions = typeof LoansActions.loadLoanDetailDocSuccess;
		function noExistingLoanDetail(action: LoansActions, loansInitialState, loanDetail: LoanDetailDoc) {
			const createAction = action({ loanDetail });

			const result = fromLoans.loansReducer(loansInitialState, createAction);

			expect(result).toMatchSnapshot();
		}

		it('should should return the full loan detail for requested loan number', () => {
			noExistingLoanDetail(LoansActions.loadLoanDetailDocSuccess, loansInitialState, mockLoanDetailDoc);
		});
	});


	describe('query loans', () => {
		it('should set the loan query on the state', () => {
			const action = LoansActions.queryLoans({ query: mockLoanQuery });

			const result = fromLoans.loansReducer(loansInitialState, action);

			expect(result).toMatchSnapshot();
		});
	});

	describe('LOAD_LOANS', () => {
		const expectedResult = {
			ids: [],
			entities: {},
			selectedLoanId: null,
			isLoading: true,
			loaded: false,
			loanQuery: null,
			loanDetail: null
		};
	});

	describe('SELECT', () => {
		it('should set the selected loan number on the state', () => {
			const action = LoansActions.selectLoan({ loanNumber: loan1.loanNumber });

			const result = fromLoans.loansReducer(loansInitialState, action);

			expect(result).toMatchSnapshot();
		});
	});

	describe('Selectors', () => {
		describe('selectId', () => {
			it('should return the selected id', () => {
				const result = fromLoans.getSelectedLoanId({
					...loansInitialState,
					selectedLoanId: loan1.loanNumber
				});

				expect(result).toMatchSnapshot();
			});
		});
	});
});
