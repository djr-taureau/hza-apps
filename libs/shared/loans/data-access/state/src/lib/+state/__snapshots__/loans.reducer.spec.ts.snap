// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoansReducer LOANS_LOAD_SUCCESS should add all loans in the payload when none exist 1`] = `
Object {
  "entities": Object {
    "12": Object {
      "City": "Dallas",
      "LoanNumber": "3404040",
      "Name": "Mary Johnson",
      "Phone": "2142239999",
      "SourceID": 1,
      "id": 12,
    },
    "2": Object {
      "City": "Dallas",
      "LoanNumber": "3404040",
      "Name": "Mary Johnson",
      "Phone": "2142239999",
      "SourceID": 1,
      "id": 2,
    },
  },
  "ids": Array [
    2,
    12,
  ],
  "isLoading": false,
  "loaded": true,
  "selectedLoanId": null,
}
`;

exports[`LoansReducer SELECT should set the selected loan number on the state 1`] = `
Object {
  "entities": Object {
    "12": Object {
      "City": "Dallas",
      "LoanNumber": "3404040",
      "Name": "Mary Johnson",
      "Phone": "2142239999",
      "SourceID": 1,
      "id": 12,
    },
    "2": Object {
      "City": "Dallas",
      "LoanNumber": "3404040",
      "Name": "Mary Johnson",
      "Phone": "2142239999",
      "SourceID": 1,
      "id": 2,
    },
  },
  "ids": Array [
    2,
    12,
  ],
  "isLoading": false,
  "loaded": false,
  "selectedLoanId": 2,
}
`;

exports[`LoansReducer Selectors selectId should return the selected id 1`] = `2`;

exports[`LoansReducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "isLoading": false,
  "loaded": false,
  "selectedLoanId": null,
}
`;
